dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,277 @@
+
+    def from_db_type(
+        self, db_type: str, precision: Optional[int] = None, scale: Optional[int] = None
+    ) -> TColumnType:
+        if db_type == "Decimal" and (precision, scale) == self.capabilities.wei_precision:
+            return dict(data_type="wei")
+        return super().from_db_type(db_type, precision, scale)
+
+
+class ClickhouseLoadJob(LoadJob, FollowupJob):
+    def __init__(
+        self,
+        file_path: str,
+        table_name: str,
+        client: ClickhouseSqlClient,
+        staging_credentials: Optional[CredentialsConfiguration] = None,
+    ) -> None:
+        file_name = FileStorage.get_file_name_from_file_path(file_path)
+        super().__init__(file_name)
+
+        qualified_table_name = client.make_qualified_table_name(table_name)
+
+        bucket_path: str = (
+            NewReferenceJob.resolve_reference(file_path)
+            if NewReferenceJob.is_reference_job(file_path)
+            else ""
+        )
+        file_name = (
+            FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+        )
+        file_extension = os.path.splitext(file_name)[1].lower()
+        if file_extension not in ["parquet", "jsonl"]:
+            raise ValueError("Clickhouse staging only supports 'parquet' and 'jsonl' file formats.")
+
+        if not bucket_path:
+            # Local filesystem.
+            raise NotImplementedError("Only object storage is supported.")
+
+        bucket_url = urlparse(bucket_path)
+        bucket_scheme = bucket_url.scheme
+
+        table_function: str
+
+        if bucket_scheme in ("s3", "gs", "gcs"):
+            bucket_http_url = convert_storage_to_http_scheme(bucket_url)
+
+            table_function = (
+                render_s3_table_function(
+                    bucket_http_url,
+                    staging_credentials.aws_secret_access_key,
+                    staging_credentials.aws_secret_access_key,
+                    file_format=file_extension,  # type: ignore[arg-type]
+                )
+                if isinstance(staging_credentials, AwsCredentialsWithoutDefaults)
+                else render_s3_table_function(
+                    bucket_http_url,
+                    file_format=file_extension,  # type: ignore[arg-type]
+                )
+            )
+        elif bucket_scheme in ("az", "abfs"):
+            if isinstance(staging_credentials, AzureCredentialsWithoutDefaults):
+                # Authenticated access.
+                account_name = staging_credentials.azure_storage_account_name
+                storage_account_url = (
+                    f"{staging_credentials.azure_storage_account_name}.blob.core.windows.net"
+                )
+                account_key = staging_credentials.azure_storage_sas_token
+                container_name = bucket_url.netloc
+                blobpath = bucket_url.path
+
+                format_mapping = {"jsonl": "JSONEachRow", "parquet": "Parquet"}
```

Comment:
this mapping is done in two different places, maybe make it more DRY

Reply:
Fixed in 86685fd684bff29c8c016d9f20bfb4b5d9055e8d